### ISSUE TITLE
Semantically-hidden and inert landmark regions

### ISSUE THREAD

**Issue by matatk** [matatk/landmarks#198]
As per https://github.com/matatk/landmarks/issues/80#issuecomment-413165045, we wish to discourage web authors from applying `aria-hidden` to an otherwise visually-apparent landmark region (it seems that hiding what is expected to be a big part of the page from assistive technology users, but not hiding it visually, is almost certainly likely to be a bug/misunderstanding).

Landmarks doesn't check for `aria-hidden`, so de facto ignores it, so such landmarks would be surfaced. What's needed is a warning to explain that they may be hidden from some users and why. Again https://github.com/matatk/landmarks/issues/80#issuecomment-413165045 has more rationale and suggestions.

**Comment by matatk** [comment]
Could make the border of these landmarks dashed, as well as including a short explanatory phrase in the label (and a warning icon in the UI tree).

**Comment by jnurthen** [comment]
This is very common when a dialog is open - the base page is hidden using aria-hidden.

**Comment by matatk** [comment]
@jnurthen that's a good point; thanks. When I said that this should be discouraged, I was specifically meaning the case where there's an author error—I've seen in the wild things like `<section aria-labelledby="..." aria-hidden="true">` (even when no dialog is open).

However, we were to encounter code such as `<main aria-hidden="true">` for example it sounds more likely to be valid, as one could imagine a dialog being a sibling element to the main page content.

There's also the `inert` attribute to consider, too.

Currently, Landmarks will show all of the detected landmark regions on the page, [unless they have the HTML5 `hidden` attribute, or are otherwise visually hidden](https://github.com/matatk/landmarks/blob/cfbfd6cdc3361b8f08b67726aec4f35dd2c23733/src/code/landmarksFinder.js#L243-L249). It *will* still surface landmark regions that are either `aria-hidden` directly, or are the children of such an element.

This can be demonstrated on [this page with example modal dialogs on it](https://scottaohara.github.io/accessible_modal_window/).

I'm now wondering two things:

1. It actually seems best for everyone to hide/disregard landmarks that are descendents of `aria-hidden` or `inert` elements. When a dialog opens, all other landmarks behind it are irrelevant (or should be). If the dialog contains any landmarks, they'd be shown.

   Whilst that seems logical to me, I've not yet looked at what the various specifciations have to say about semantically-hidden landmark regions, so will check them out.

2. In the case of the specific author error, if that can be reasonably detected, then it sounds like it should be added as one of the [proposed linting errors](https://github.com/matatk/landmarks/issues/182#issuecomment-418528952).

Thanks again for your comment, it has really made me think :-).

**Comment by matatk** [comment]
I've [updated the test suite](https://github.com/matatk/page-structural-semantics-scanner-tests/blob/master/CHANGELOG.md#050-2020-04-24) and closed this issue in #360.

Test cases:
* https://scottaohara.github.io/accessible_modal_window/
* https://van11y.net/downloads/modal/demo/index.html
* https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/dialog.html (which has highlighted that `aria-modal` is another consideration, to be addressed separately)
